saleor/search/views.py

```diff
@@ -0,0 +1,36 @@
+import datetime
+
+from django.core.paginator import Paginator, InvalidPage
+from django.conf import settings
+from django.http import Http404
+from django.shortcuts import render
+from haystack.forms import SearchForm
+
+from ..product.models import Product
+
+
+def paginate_results(results, get_data, paginate_by=25):
+    paginator = Paginator(results, paginate_by)
+    page_number = get_data.get('page', 1)
+    try:
+        page = paginator.page(page_number)
+    except InvalidPage:
+        raise Http404("No such page!")
+    return page
+
+
+def search(request):
+    form = SearchForm(data=request.GET or None, load_all=True)
+    today = datetime.date.today()
+    if form.is_valid():
+        results = form.search().models(Product)
+        results = results.filter_or(available_on__lte=today)
```

Comment:
We should not push business logic here. Similarly to how product lists are handled, please create a function that does the right thing here.